src/ring_theory/finiteness.lean

```diff
@@ -90,7 +90,24 @@ variables (R)
 instance self : finite R R :=
 ⟨⟨{1}, by simpa only [finset.coe_singleton] using ideal.span_singleton_one⟩⟩
 
-variables {R}
+variable (M)
+
+lemma of_restrict_scalars_finite [module A M] [is_scalar_tower R A M] [hM : finite R M] :
+  finite A M :=
+begin
+  rw [finite_def, fg_def] at hM ⊢,
+  obtain ⟨S, hSfin, hSgen⟩ := hM,
+  refine ⟨S, hSfin, eq_top_iff.2 _⟩,
+  have := submodule.span_le_restrict_scalars R A M S,
+  rw hSgen at this,
+  exact this
+end
+
+lemma of_restrict_scalars_finite' (M : Type*) [add_comm_group M] [module A M]
+  (hM : finite R ((restrict_scalars R A M))) : finite A M :=
+by convert of_restrict_scalars_finite R A (restrict_scalars R A M)
+
```

Comment:
To be clear on my comment above; I'm mildly in favor of just removing the two `restrict_scalars R A M` lemmas you added, I can't seen them ever being used. If you or another maintainer think they're useful then feel free to keep them, but don't keep them just because of my confused comments earlier!